installer/src/App.jsx

```diff
@@ -95,33 +115,43 @@ function App() {
             return;
         }
 
-        const uploadMsg = `Uploading binary ${dstPath.split(/.*[/|\\]/)[1]} to ${board.port}\n`;
-        dispatch({ type: 'APPEND_STDOUT', stdout: uploadMsg });
+        dispatch({
+            type: 'APPEND_STDOUT',
+            stdout: 'Checking installed Arduino cores',
+        });
 
-        const onData = (line) =>
-            dispatch({ type: 'APPEND_STDOUT', stdout: line });
+        try {
+            await installRequiredCores({
```

Comment:
I think we could just install the cores that is needed fo the current board instead of everyone. This check could be executed right before the upload and based on the fqbn installing the code. I would also integrate the check in the upload itself.